File: app/src/main/java/com/skybox/seven/covid/ui/contactTracing/ContactRequestFragment.java
package com.skybox.seven.covid.ui.contactTracing;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import com.airbnb.epoxy.EpoxyRecyclerView;
import com.skybox.seven.covid.R;
import com.skybox.seven.covid.epoxy.contacts.ContactsRequestController;
import com.skybox.seven.covid.util.InjectorUtil;
import com.skybox.seven.covid.ui.contactTracing.ContactsViewModel;

/**
 * A simple {@link Fragment} subclass.
 */
public class ContactRequestFragment extends Fragment implements ContactsRequestController.ContactsCallBack {

    private ContactsRequestController controller;
    private ContactsViewModel viewModel;
    private SwipeRefreshLayout refreshLayout;

    public ContactRequestFragment() {
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_generic_epoxy_swipe, container, false);
        EpoxyRecyclerView recyclerView = v.findViewById(R.id.generic_recycler_id);
        refreshLayout = v.findViewById(R.id.generic_swipe_view);

        viewModel = new ViewModelProvider(getActivity(), InjectorUtil.provideContactsViewModelFactory(getContext())).get(ContactsViewModel.class);
        controller = new ContactsRequestController(this);

        recyclerView.setController(controller);

        refreshLayout.setOnRefreshListener(() -> viewModel.generatePendingContactList());

        viewModel.contactsRefresh.observe(getActivity(), aBoolean -> viewModel.getAllContacts());
        viewModel.pendingContacts.observe(getViewLifecycleOwner(), pendingContacts -> controller.setData(false,pendingContacts));
        viewModel.contactsLoading.observe(getViewLifecycleOwner(), aBoolean ->{
            controller.setData(aBoolean, viewModel.pendingContacts.getValue());
            if (!aBoolean) {
                refreshLayout.setRefreshing(false);
            }
        });

        viewModel.generatePendingContactList();
        return v;
    }


    @Override
    public void onContactAcceptClick(int id, int position) {
        viewModel.acceptContact(id, position);
    }

    @Override
    public void onContactRejectClick(int id, int position) {
        viewModel.rejectContact(id, position);
    }
}

File: app/src/main/java/com/skybox/seven/covid/ui/profile/contacts/pages/ContactRequestFragment.kt
package com.skybox.seven.covid.ui.profile.contacts.pages

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import com.skybox.seven.covid.databinding.FragmentGenericEpoxySwipeBinding
import com.skybox.seven.covid.epoxy.contacts.ContactsRequestController
import com.skybox.seven.covid.epoxy.contacts.ContactsRequestController.ContactsCallBack
import com.skybox.seven.covid.model.ContactRequestModel.PendingContacts
import com.skybox.seven.covid.ui.profile.contacts.ContactsViewModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class ContactRequestFragment : Fragment(), ContactsCallBack {
    private lateinit var binding: FragmentGenericEpoxySwipeBinding

    private var controller: ContactsRequestController = ContactsRequestController(this)
    private val viewModel: ContactsViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel.contactsRefresh.observe(this, Observer { viewModel.allContacts })
        viewModel.pendingContacts.observe(this, Observer<List<PendingContacts?>> { pendingContacts: List<PendingContacts?> -> controller.setData(false, pendingContacts) })
        viewModel.generatePendingContactList()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        binding = FragmentGenericEpoxySwipeBinding.inflate(inflater, container, false)

        binding.genericRecyclerId.setController(controller)
        binding.genericSwipeView.setOnRefreshListener { viewModel.generatePendingContactList() }

        viewModel.contactsLoading.observe(viewLifecycleOwner, Observer { aBoolean: Boolean ->
            controller.setData(aBoolean, viewModel.pendingContacts.value)
            if (!aBoolean) {
                binding.genericSwipeView.isRefreshing = false
            }
        })
        return binding.root
    }

    override fun onContactAcceptClick(id: Int, position: Int) {
        viewModel.acceptContact(id, position)
    }

    override fun onContactRejectClick(id: Int, position: Int) {
        viewModel.rejectContact(id, position)
    }
}